DataFormats/EcalDigi/BuildFile.xml

```diff
@@ -3,6 +3,7 @@
 <use name="DataFormats/DetId"/>
 <use name="FWCore/MessageLogger"/>
 <use name="FWCore/Utilities"/>
+<use name="HeterogeneousCore/CUDAUtilities"/>
```

Comment:
Unfortunately `DataFormats` package depending on `HeterogeneousCore` is not allowed (in order to avoid dependence on CUDA in persistent data formats).

The header `HostDeviceConstant.h` itself depends on CUDA only "weakly" though (it only uses macros that are defined by CUDA-capable compiler and as such it does not require CUDA by itself). Maybe that header should be placed somewhere in `DataFormats` instead? (assuming `EcalConstants.h` can not be moved to non-dataformat package)

Reply:
What about moving it to `FWCore/Utilities`, like [FWCore/Utilities/interface/CMSUnrollLoop.h](https://github.com/cms-sw/cmssw/blob/master/FWCore/Utilities/interface/CMSUnrollLoop.h) ?

Reply:
That would work too.

Reply:
OK, I've updated #37159 accordingly.

Reply:
I have updated this one as well.